admin_ui/app/org/OrgEditPermissionsModals.js

```diff
@@ -0,0 +1,117 @@
+import React, { useContext, useEffect } from 'react';
+import { useObservable } from 'rxjs-hooks';
+import Store from '../Store';
+import { Link } from '@reach/router';
+import Modal from '../../components/Modal';
+import Input from '../../components/Input';
+import Button from '../../components/Button';
+import Textarea from '../../components/Textarea';
+
+// The CREATE, EDIT or DELETE modals for the OrgEditPermissions page
+const OrgEditPermissionsModals = () => {
+  // Load the store
+  const store = useContext(Store);
+
+  // Establish the value of the currentModal$ observable
+  const currentModal = useObservable(
+    () => store.org.currentPermissionsModal$,
+    store.org.currentPermissionsModal$.getValue()
+  );
+
+  const handleESC = (e) => {
+    if (e.key === "Escape") {
+      // ESC key pressed. Hide any modals
+      store.org.currentPermissionsModal$.next('');
+    }
+  };
+
+  // Hide any modal on ESC keypress
+  // This will be an effect (with cleanup).
+  // We pass [] as second argument to avoid multiple add + remove invocaitons
+  // https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects for details
+  useEffect(() => {
+    // console.log('useEffect called!!!');
+    window.document.addEventListener('keydown', handleESC);
+    return () => {
+      // console.log('useEffect: removing listener');
+      window.removeEventListener('keydown', handleESC);
+    };
+  }, []);
+
+  if (currentModal === 'CREATE') {
```

Comment:
I'd personally take each one of those modals and create a separate component (in separate modules or this particular module), then use a switch statement to render the correct component.

Reply:
Agreed! A good optimisation. I opted to move them into a separate folder for ease of parsing:

![image](https://user-images.githubusercontent.com/14941382/57578083-59116100-748e-11e9-9ed1-b10a421d08ee.png)

I also left the `cost store = useContext(Store)` invocation inside each of those. I'll _probably_ move this to the parent `OrgEditPermissionsModals.js` and `OrgEditRolesModals.js` components when I start implementing actual business logic / data fetching.